instrumentation/grpc-1.30.0/src/main/java/io/grpc/ClientCall_Instrumentation.java

```diff
@@ -0,0 +1,131 @@
+/*
+ *
+ *  * Copyright 2020 New Relic Corporation. All rights reserved.
+ *  * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+package io.grpc;
+
+import com.newrelic.agent.bridge.AgentBridge;
+import com.newrelic.api.agent.ExternalParameters;
+import com.newrelic.api.agent.HttpParameters;
+import com.newrelic.api.agent.NewRelic;
+import com.newrelic.api.agent.Segment;
+import com.newrelic.api.agent.weaver.MatchType;
+import com.newrelic.api.agent.weaver.NewField;
+import com.newrelic.api.agent.weaver.Weave;
+import com.newrelic.api.agent.weaver.Weaver;
+import com.nr.agent.instrumentation.grpc.GrpcConfig;
+import com.nr.agent.instrumentation.grpc.InboundHeadersWrapper;
+import com.nr.agent.instrumentation.grpc.OutboundHeadersWrapper;
+import io.grpc.stub.ClientCalls_Instrumentation;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.logging.Level;
+
+@Weave(type = MatchType.BaseClass, originalName = "io.grpc.ClientCall")
+public class ClientCall_Instrumentation<ReqT, RespT> {
+
+    @NewField
+    public Segment segment = null;
+
+    @NewField
+    public String authority = null;
+
+    @NewField
+    public MethodDescriptor<ReqT, RespT> methodDescriptor;
+
+    public void start(Listener<RespT> responseListener, Metadata headers) {
+        if (segment != null) {
+            responseListener.segment = segment;
```

Comment:
these assignments are new. Those fields on the responseListener existed in grpc 1.22 instrumentation. Should those fields in 1.22 be updated as well then?

Reply:
The old instrumentation had access to the listener in the stub ClientCalls class, and set them [there](https://github.com/newrelic/newrelic-java-agent/blob/main/instrumentation/grpc-1.22.0/src/main/java/io/grpc/stub/ClientCalls_Instrumentation.java#L29-L31https://github.com/newrelic/newrelic-java-agent/blob/main/instrumentation/grpc-1.22.0/src/main/java/io/grpc/stub/ClientCalls_Instrumentation.java#L29-L31).